IPython/core/guarded_eval.py

```diff
@@ -654,27 +693,50 @@ def eval_node(node: Union[ast.AST, None], context: EvaluationContext):
                 continue
             if decorator is property:
                 is_property = True
+            elif decorator is staticmethod:
+                is_static = True
+            elif decorator is classmethod:
+                is_classmethod = True
+
+        if func_context.class_transients is not None:
+            if not is_static and not is_classmethod:
+                func_context.instance_arg_name = node.args.args[0].arg
```

Comment:
Completion breaks if an instance method is defined incorrectly:

<img width="248" height="120" alt="Image" src="https://github.com/user-attachments/assets/d54a81e7-9587-4a21-b56f-fe6d5b024e35" />

Works fine if there is an argument:

<img width="289" height="120" alt="Image" src="https://github.com/user-attachments/assets/1daf6e09-1368-46b7-99f2-31ab3e5ea171" />

Since only one incorrectly defined method breaks completion for the entire class I suggest we first check if there are any arguments here. Let's also add a test case.